### DISCUSSION TITLE
Fastlane match hangs with private repo

### DISCUSSION THREAD

**Discussion by serhatsezer** [fastlane/fastlane#17577] [Q&A]
**Problem Description**

I'm trying to use Match to handle my certificates and other stuff with my private gitlab account. I've already created private repo and then added ssh url to my Match file. But whenever I try to run `fastlane match adhoc` it hangs and never proceeds. I'm trying to solve this for about a week so any helps very much appreciated 🙏🏻

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                   |
| --------------------------- | ------------------------------------------------------- |
| OS                          | 10.15.7                                                 |
| Ruby                        | 2.7.2                                                   |
| Bundler?                    | false                                                   |
| Git                         | git version 2.24.3 (Apple Git-128)                      |
| Installation Source         | /usr/local/Cellar/fastlane/2.163.0/libexec/bin/fastlane |
| Host                        | Mac OS X 10.15.7 (19H2)                                 |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/2.7.2/lib                        |
| OpenSSL Version             | OpenSSL 1.1.1g  21 Apr 2020                             |
| Is contained                | false                                                   |
| Is homebrew                 | false                                                   |
| Is installed via Fabric.app | false                                                   |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/             |
| Xcode Version               | 12.1                                                    |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
default_platform(:ios)

platform :ios do
  desc "Producing the app"
  lane :register_app do
  	produce(username: "myaccount@icloud.com", 
  		app_identifier: "com.sample.cdsetup", 
  		app_name: "CdSetupExample",
  		team_name: "Serhat Sezer",
  		itc_team_name: "Serhat Sezer")
  end

  lane :sandbox do 
  	
  end

  lane :get_dev_certs do 
  	cert(development: true)
  	sigh(development: true)
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# app_identifier("[[APP_IDENTIFIER]]") # The bundle identifier of your app
# apple_id("[[APPLE_ID]]") # Your Apple email address

app_identifier "com.sample.cdsetup"
apple_id "myaccount@icloud.com"
team_id "122315809"
# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.165.0 | 🚫 Update available |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.4.0        |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.3        |
| CFPropertyList            | 3.0.2        |
| claide                    | 1.0.3        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.3.0        |
| xcodeproj                 | 1.19.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.7.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| public_suffix             | 4.0.6        |
| addressable               | 2.7.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.8.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| babosa                    | 1.0.4        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.7        |
| faraday_middleware        | 1.0.0        |
| fastimage                 | 2.2.0        |
| gh_inspector              | 1.1.3        |
| mini_magick               | 4.10.1       |
| rubyzip                   | 2.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.7.6        |
| bundler                   | 2.1.4        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| jwt                       | 2.2.2        |
| uber                      | 0.1.0        |
| declarative               | 0.0.20       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| multi_json                | 1.15.0       |
| signet                    | 0.14.0       |
| memoist                   | 0.16.2       |
| os                        | 1.1.1        |
| googleauth                | 0.14.0       |
| httpclient                | 2.8.3        |
| google-api-client         | 0.38.0       |
| google-cloud-errors       | 1.0.1        |
| google-cloud-core         | 1.5.0        |
| rake                      | 13.0.1       |
| digest-crc                | 0.6.1        |
| google-cloud-storage      | 1.29.1       |
| uri                       | 0.10.0       |
| excon                     | 0.78.0       |
| ruby2_keywords            | 0.0.2        |
| faraday                   | 1.1.0        |
| json                      | 2.3.1        |
| google-cloud-env          | 1.4.0        |
| emoji_regex               | 3.2.1        |
| aws-eventstream           | 1.1.0        |
| aws-sigv4                 | 1.2.2        |
| aws-partitions            | 1.388.0      |
| jmespath                  | 1.4.0        |
| aws-sdk-core              | 3.109.1      |
| aws-sdk-kms               | 1.39.0       |
| aws-sdk-s3                | 1.83.1       |
| forwardable               | 1.3.1        |
| logger                    | 1.4.2        |
| cgi                       | 0.1.0        |
| timeout                   | 0.1.0        |
| stringio                  | 0.1.0        |
| ipaddr                    | 1.2.2        |
| openssl                   | 2.1.2        |
| ostruct                   | 0.2.0        |
| strscan                   | 1.0.3        |
| date                      | 3.0.0        |
| delegate                  | 0.1.0        |
| fileutils                 | 1.4.1        |
| io-console                | 0.5.6        |
| zlib                      | 1.1.0        |
| singleton                 | 0.1.0        |
| rexml                     | 3.2.3        |
| open3                     | 0.1.0        |
| yaml                      | 0.1.0        |
| psych                     | 3.1.0        |
| mutex_m                   | 0.1.0        |
| webrick                   | 1.6.0        |
</details>

*generated on:* **2020-11-09**
</details>

**Comment by rogerluan** [comment]
Hey @serhatsezer 👋 

Can you provide your `Matchfile`?

**Reply by serhatsezer** [reply]
Hi @rogerluan,

Thank you for reply. I use lane instead of Matchfile. 
Here is my lane:

```ruby
  desc "Fetch AdHoc profiles"
  lane :get_adhoc_profiles do
    match(
      git_url: "git@gitlab.com:company/company/match-address.git",
      type: "adhoc",
      app_identifier: ["com.company.app", "com.company.app.widget"],
      readonly: true,
      keychain_name: ENV["MATCH_KEYCHAIN_NAME"],
      keychain_password: ENV["MATCH_KEYCHAIN_PASSWORD"]
    )
  end
```

**Reply by rogerluan** [reply]
Hey @serhatsezer , I couldn't spot anything odd from your configuration. Being a private repo shouldn't be a problem on the _fastlane_ end, but did you certify that your machine (local or CI, idk which one you're using) has access to that private repo? In other words, is the ssh configured correctly? You can test that by attempting to run e.g. `git clone git@gitlab.com:company/company/match-address.git` directly and see the results, for instance. 
Also I've only personally used match using GitHub, not GitLab, but that shouldn't make any difference AFAIK.

Let me know how your tests goes 🙏

**Reply by serhatsezer** [reply]
Hi @rogerluan, Thank you for your comment. I still couldn't be able to make it work. I successfully pull the files just running recommended comment but in CI it throws auth error. 

```ruby desc "Fetch AdHoc profiles"
  lane :get_adhoc_profiles do
    match(
      git_url: "https://gitlab.com/project/project/private-repo-match.git",
      git_basic_authorization: Base64.encode64("mygitlabusername:my_personal_auth_key"),
      type: "adhoc",
      app_identifier: ["com.bundle.app", "com.bundle.app.widget"],
      readonly: true
    )
  end
```

**Reply by rogerluan** [reply]
@serhatsezer does that same lane work on your local machine?
Does your CI happen to be Azure DevOps? 
In [the docs](https://docs.fastlane.tools/actions/match/#git-storage) there're a couple examples of how to setup your auth, depending on whether you're using SSH, basic auth, deploy keys, Azure DevOps, etc - and each of them is different from the other. Did you check those for troubleshooting?
